lib/matplotlib/figure.py

```diff
@@ -964,13 +965,44 @@ def set_facecolor(self, color):
         """
         self.patch.set_facecolor(color)
 
-    def set_dpi(self, val):
+    def set_dpi(self, dpi):
         """
-        Set the resolution of the figure in dots-per-inch.
+        Set the nominal resolution of the figure in dots-per-inch.
 
-        .. ACCEPTS: float
-        """
-        self.dpi = val
+        This is nominal because some screens have "hi-dpi" or "Retina",
+        where pixels are doubled.  This dpi is the non-doubled value.
+
+        Setting the dpi different from your screen dpi makes the figure larger
+        or smaller on the screen than the width and height specified in
+        `~.Figure.get_width_height`.
+
+        Parameters
+        ----------
+        dpi : float
+
+        Notes
+        -----
+        Some backends have the concept of "hi-dpi" displays (or "Retina")
+        where the resolution is doubled, but dimensions are traditionally
+        specified as if the resolution were not doubled.  Some Matplotlib
+        GUI backends respect this doubling (i.e. Qt5Agg), but Matplotlib still
+        specifies dimensions and dpi to matplotlib as if the resolution
+        were not doubled. The default figure
+        value is nominally 100 dpi, but displays that are deignated "hi-dpi"
+        will internally double this to 200 dpi.  We keep the "nominal" dpi
```

Comment:
This is not quite true, we also need to save the original to drop back to the correct dpi for savefig with `dpi='figure'`.